### PR TITLE
Task J-223: + sort for time navigation

### PR DESCRIPTION
**Pull request by JuliaDzyuba**

### CODE REVIEW

File: frontend/src/components/episode/episode.tsx [hunk 1]
```diff
@@ -166,7 +167,7 @@ const Episode: React.FC = () => {
                   <div className={styles.shownotesWrapper}>
                     <h3>Time navigation</h3>
                     <ShownotesList
-                      shownotes={episode.shownotes}
+                      shownotes={getSortedShownotes(episode?.shownotes)}
```
Comment (Denidzio) [inline frontend/src/components/episode/episode.tsx]: Would it be better to sort directly in the 'shownote-list' component?

Comment (JuliaDzyuba) [reply frontend/src/components/episode/episode.tsx]: May be, but I'm not sure

Comment (what1s1ove) [reply frontend/src/components/episode/episode.tsx]: No, shownote-lists shouldn't know anything about sort

File: frontend/src/components/episode/helpers/sort-shownotes/sort-shownotes.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { Shownote } from 'common/types/types';
+
+const getSortedShownotes = ( shownotes: Shownote[]): Shownote[] => {
+  return [...shownotes].sort((a, b) => a.timestamp - b.timestamp);
```
Comment (what1s1ove) [inline frontend/src/components/episode/helpers/sort-shownotes/sort-shownotes.ts]: we have the similar functionality in several places

let's reuse it and create a `getSortedItems` helper

Something like

```ts
const getSortedItems = <T>(items: T[], (itemA: T, itemB: T) => number): T[] => {
  // your logic
}
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Comment by JuliaDzyuba** [comment]
maybe should add an enum for the object keys and an enum for the sorting type?

File: shared/src/helpers/array/get-sorted-items/get-sorted-items.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+const getSortedItems = <T>(
+  items: T[],
+  keyToSort: keyof T,
```
Comment (what1s1ove) [inline shared/src/helpers/array/get-sorted-items/get-sorted-items.ts]: we need to pass sort cb for it

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: shared/src/helpers/array/get-sorted-items/get-sorted-items.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+
+const getSortedItems = <T>(
+  items: T[],
+  cb: (itemA: T, itemB: T) => number,
+): T[] => {
+  return [...items].sort(cb);
```
Comment (what1s1ove) [inline shared/src/helpers/array/get-sorted-items/get-sorted-items.ts]:
```suggestion
  return items.slice().sort(cb);
```

File: frontend/src/components/episode/helpers/sort-shownotes/sort-shownotes.ts (hunk 2 above)
Comment (what1s1ove) [inline frontend/src/components/episode/helpers/sort-shownotes/sort-shownotes.ts]:
```suggestion
  return getSortedItems(shownotes, (a, b) => a.timestamp - b.timestamp);
```

File: frontend/src/components/episode/episode.tsx [hunk 5]
```diff
@@ -101,6 +101,8 @@ const Episode: React.FC = () => {
   );
   const podcastDuration = playerRef.current?.getPodcastDuration();
 
+  const sortCB = (a: Shownote, b: Shownote): number => a.timestamp - b.timestamp;
```
Comment (what1s1ove) [inline frontend/src/components/episode/episode.tsx]:
```suggestion
  const sortedShownotes = getSortedShownotes(episode.shownotes);
```

File: frontend/src/components/episode/episode.tsx [hunk 6]
```diff
@@ -166,7 +168,7 @@ const Episode: React.FC = () => {
                   <div className={styles.shownotesWrapper}>
                     <h3>Time navigation</h3>
                     <ShownotesList
-                      shownotes={episode.shownotes}
+                      shownotes={getSortedItems(episode.shownotes, sortCB)}
```
Comment (what1s1ove) [inline frontend/src/components/episode/episode.tsx]:
```suggestion
                      shownotes={sortedShownotes}
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**